ement-room.el

```diff
@@ -2817,151 +2809,113 @@ MSG is the message STRING was contained in."
-;;   "Insert link preview for DATA into MSG, if it isn't already present.
-;; BUF is the room buffer this link is from. NUM is the index for this link."
-;;   (cond ((alist-get 'errcode data) (message "Ement: You are being rate limited."))
-;;         (data (with-current-buffer buf
-;;                 (let* ((node (ement-room--ewoc-node-before ement-ewoc
-;;                                                            msg
-;;                                                            (lambda (a b)
-;;                                                              (string= (ement-event-id a)
-;;                                                                       (ement-event-id b)))
-;;                                                            :pred #'ement-event-p))
-;;                        (next? (cl-loop with w = (ewoc-next ement-ewoc node)
-;;                                        repeat num
-;;                                        if (or (not w)
-;;                                               (not (listp (ewoc-data w)))
-;;                                               (not (eq (car (ewoc-data w))
-;;                                                        'url-preview))
-;;                                               (not (equal (cadr (ewoc-data w))
-;;                                                           data)))
-;;                                        return t
-;;                                        do (setf w (ewoc-next ement-ewoc w))
-;;                                        finally return nil)))
-;;                   (when next?
-;;                     (with-silent-modifications
-;;                       (ewoc-enter-after ement-ewoc
-;;                                         node
-;;                                         `(url-preview ,data)))))))))
-
-(defun ement-room--insert-preview (buf msg related data num)
+(defun ement-room--get-preview (msg link num)
+  "Get preview of LINK. MSG is the message this link is from. NUM is the index for this link."
+  (let* ((id (ement-event-id msg))
+         (related-event (cl-loop with match = (make-ement-event :id id)
+                                 for e in (ement-room-timeline ement-room)
+                                 if (ement--events-equal-p match e)
+                                 return e))
+         (saved-buffer (current-buffer)))
+    (unless (or ement-room--preview-stop
+                (map-elt (ement-event-local related-event)
+                         'url-preview-n))
+      (ement-api ement-session "preview_url"
+        :endpoint-category "media" :params `((url ,link))
+        :then (lambda (data)
+                (when (buffer-live-p saved-buffer)
+                  (with-current-buffer saved-buffer
+                    (ement-room--insert-preview msg related-event num (ement-room--cleanse-url-preview data link)))))
+        :else (lambda (why) (ement-debug "Error in fetching preview: " why))))))
+
+(defun ement-room--cleanse-url-preview (data link)
```

Comment:
For a function this small that is only called in one place, I'd prefer it be inlined.

Also, please leave a brief comment that explains why it's necessary.

Reply:
It's to have a consistent preview, some previews may be missing the simplest of basic data like a URL or a title, which results in a preview that looks weird (eg: A preview with only a description, or maybe a preview with nothing at all).

Basically to ensure that all previews at least have a title and url associated with it. I should probably just inline the operations as you suggested though, since it's so small.